config/routes.rb

```diff
@@ -426,6 +426,9 @@
 
     post 'amr_data_feed_readings/:amr_uploaded_reading_id', to: 'amr_data_feed_readings#create', as: :create_amr_data_feed_readings
 
+    authenticated :user, -> user { user.admin? }  do
```

Comment:
Could you add a simple system spec to confirm that the authentication of the route is working as expected? Maybe one for a guest/school user and one for admin.

Reply:
:+1:   have just pushed an update with some tests.